Members data source crashes

```
terraform {
  required_providers {
    zerotier = {
      source  = "zerotier/zerotier"
      version = "~> 1.5"
    }
  }
}

provider "zerotier" {
  zerotier_central_token = var.token
}

variable "token" {
  type = string
}

variable "network_id" {
  type = string
}

data zerotier_network this {
  id = var.network_id
}

data zerotier_members this {
  network_id = data.zerotier_network.this.id
}

output "network_name" {
  value = data.zerotier_network.this.name
}

output "members" {
  value = data.zerotier_members.this.members
}
```

```
$ terraform apply
data.zerotier_network.this: Reading...
data.zerotier_network.this: Read complete after 2s [id=xxxxxxxxxxxxxxxx]
data.zerotier_members.this: Reading...

Changes to Outputs:
  + network_name = "somenetwork"

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.
╷
│ Error: Plugin did not respond
│
│   with data.zerotier_members.this,
│   on main.tf line 26, in data "zerotier_members" "this":
│   26: data zerotier_members this {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-zerotier_1.5.1 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 74 [running]:
github.com/zerotier/terraform-provider-zerotier/pkg/zerotier.resourceNetworkAndNodeIdentifiers(0x14000162300)
	github.com/zerotier/terraform-provider-zerotier/pkg/zerotier/resource_member.go:92 +0x188
github.com/zerotier/terraform-provider-zerotier/pkg/zerotier.datasourceMemberRead({0x1013098c8, 0x140001355c0}, 0x0?, {0x1012d9600?, 0x14000241e30})
	github.com/zerotier/terraform-provider-zerotier/pkg/zerotier/data_source_members.go:38 +0x58
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1400024c380, {0x101309900, 0x14000134d20}, 0xd?, {0x1012d9600, 0x14000241e30})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:750 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0x1400024c380, {0x101309900, 0x14000134d20}, 0x14000162200, {0x1012d9600, 0x14000241e30})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:970 +0x130
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0x14000246330, {0x101309900?, 0x14000134c00?}, 0x14000156c00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1198 +0x304
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0x140004d7f40, {0x101309900?, 0x14000134450?}, 0x140006e89b0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:661 +0x368
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x1012cb6e0?, 0x140004d7f40}, {0x101309900, 0x14000134450}, 0x1400017bce0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:458 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140004b05a0, {0x10130d540, 0x14000602680}, 0x140006e50e0, 0x14000241b60, 0x101861e50, 0x0)
	google.golang.org/grpc@v1.56.2/server.go:1337 +0xc90
google.golang.org/grpc.(*Server).handleStream(0x140004b05a0, {0x10130d540, 0x14000602680}, 0x140006e50e0, 0x0)
	google.golang.org/grpc@v1.56.2/server.go:1714 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.2/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.56.2/server.go:957 +0x16c

Error: The terraform-provider-zerotier_1.5.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/data_source_test.go
+++ b/data_source_test.go
@@ -29,3 +29,24 @@ func TestDataSourceNetwork(t *testing.T) {
 
 	}
 }
+
+func TestDataSourceMembers(t *testing.T) {
+	tf := getTFTest(t)
+
+	tf.Apply("testdata/plans/data-source-members.tf")
+	tf.Refresh()
+
+	outputs := h(tf.State()["outputs"])
+
+	memberName := s(h(outputs["member_name"])["value"])
+	memberNameExp := "bobs_car"
+	if memberName != memberNameExp {
+		t.Fatalf("member name was not set correctly: expected: %q, found: %q", memberNameExp, memberName)
+	}
+
+	memberDesc := s(h(outputs["member_description"])["value"])
+	memberDescExp := "bobs shiny car"
+	if memberDesc != memberDescExp {
+		t.Fatalf("member description was not set correctly: expected: %q, found: %q", memberDescExp, memberDesc)
+	}
+}

--- a/pkg/zerotier/data_source_members.go
+++ b/pkg/zerotier/data_source_members.go
@@ -4,11 +4,12 @@ import (
 	"context"
 	"crypto/md5"
 	"fmt"
+	"sort"
+	"strings"
+
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/zerotier/go-ztcentral"
-	"sort"
-	"strings"
 )
 
 func dataSourceMembers() *schema.Resource {
@@ -35,10 +36,7 @@ func dataSourceMembers() *schema.Resource {
 func datasourceMemberRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*ztcentral.Client)
 
-	nwid, _, err := resourceNetworkAndNodeIdentifiers(d)
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	nwid := d.Get("network_id").(string)
 
 	networkMembers, err := c.GetMembers(ctx, nwid)
 	if err != nil {
@@ -48,10 +46,6 @@ func datasourceMemberRead(ctx context.Context, d *schema.ResourceData, m interfa
 	memberIDs := make([]string, 0, len(networkMembers))
 	for _, member := range networkMembers {
 		ipv4Assignments, ipv6Assignments := assignedIpsGrouping(*member.Config.IpAssignments)
-		_, nodeID, err := resourceNetworkAndNodeIdentifiers(d)
-		if err != nil {
-			return diag.FromErr(err)
-		}
 		members = append(members, map[string]interface{}{
 			"name":                    *member.Name,
 			"description":             *member.Description,
@@ -66,12 +60,15 @@ func datasourceMemberRead(ctx context.Context, d *schema.ResourceData, m interfa
 			"tags":                    *member.Config.Tags,
 			"ipv4_assignments":        ipv4Assignments,
 			"ipv6_assignments":        ipv6Assignments,
-			"rfc4193":                 rfc4193Address(nwid, nodeID),
-			"sixplane":                sixPlaneAddress(nwid, nodeID),
+			"rfc4193":                 rfc4193Address(nwid, *member.NodeId),
+			"sixplane":                sixPlaneAddress(nwid, *member.NodeId),
 		})
 		memberIDs = append(memberIDs, *member.NodeId)
 	}
-	d.Set("members", members)
+	err = d.Set("members", members)
+	if err != nil {
+		return diag.FromErr(err)
+	}
 	d.SetId(stringChecksum(strings.Join(memberIDs, "")))
 	return nil
 }

--- a/testdata/plans/data-source-members.tf
+++ b/testdata/plans/data-source-members.tf
@@ -16,5 +16,14 @@ resource "zerotier_member" "bobs_car" {
 }
 
 data "zerotier_members" "bobs_garage" {
+  depends_on = [zerotier_member.bobs_car]
   network_id = zerotier_network.bobs_garage.id
+}
+
+output "member_name" {
+  value = one(data.zerotier_members.bobs_garage.members)["name"]
+}
+
+output "member_description" {
+  value = one(data.zerotier_members.bobs_garage.members)["description"]
 }

